### PR TITLE
Fix tokenizer UnboundLocalError when padding is set to PaddingStrategy.MAX_LENGTH

### LINKED ISSUE #7609: Tokenizer: UnboundLocalError with PaddingStrategy MAX_LENGTH
## Environment info

     
- `transformers` version: 3.1.0
- Platform: macOS-10.15.7-x86_64-i386-64bit
- Python version: 3.8.5
- PyTorch version (GPU?): 1.6.0 (False)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

### Who can help
 
 tokenizers: @mfuntowicz

## Information

Model I am using: <transformers.tokenization_bert.BertTokenizer>

The problem arises when using:
* the official example scripts: 
using the `encode_plus`

The tasks I am working on is:
* Tokenizing

## To reproduce

1. tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
2. tokenizer.encode_plus("hello word", max_length=128, padding=PaddingStrategy.MAX_LENGTH)

`UnboundLocalError: local variable 'padding_strategy' referenced before assignment`

## Expected behavior

Return Tokenizer output

### PR DESCRIPTION
**Pull request by GabrielePicco**
# What does this PR do?

Fixes #7609 

## Who can review?

 tokenizers: @mfuntowicz

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Yes!

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -1781,6 +1781,8 @@ def _get_padding_truncation_strategies(
                 padding_strategy = PaddingStrategy.LONGEST  # Default to pad to the longest sequence in the batch
             elif not isinstance(padding, PaddingStrategy):
                 padding_strategy = PaddingStrategy(padding)
+            elif isinstance(padding, PaddingStrategy):
+                padding_strategy = padding
```
Comment (thomwolf) [inline src/transformers/tokenization_utils_base.py:1785]: Ok, let's do the same for the truncation line 1809 if you don't mind to add this:
```
elif isinstance(truncation, TruncationStrategy):
                 truncation_strategy = truncation
```

Comment (GabrielePicco) [reply src/transformers/tokenization_utils_base.py:1785]: done

Comment (GabrielePicco) [reply src/transformers/tokenization_utils_base.py:1785]: @thomwolf that change break a unit test, shoud I remove the added lines ?

**Comment by LysandreJik** [comment]
Thanks!

**Pull request merged**